lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -878,6 +878,88 @@ class ConvertAtenCatOp : public OpConversionPattern<AtenCatOp> {
 };
 } // namespace
 
+// Broadcasts input tensor based on the broadcastToShape.
+template <typename OpTy>
+static LogicalResult
+broadcastToGivenShape(OpTy op, ConversionPatternRewriter &rewriter, Value input,
```

Comment:
I don't think this needs to be a template. I also noticed that there is nothing in the body of the function that requires information about the `op`. The `op` is only used to get a location and a context, however you can get a context from the rewriter itself. Here, I would use a similar signature to other helper functions

```cpp
broadcastToGivenShape(PatternRewriter &rewriter, Location loc, ...)
```

This makes it clear to the user that what this function does is `op`-independent.

Reply:
Never mind, you do need the `op` as an argument to create the `notifyMatchFailure`. So you should keep the signature unchanged. However, I would remove the template code and pass the `op` as `Operation *op`.